packages/amplify-e2e-tests/src/__tests__/auth-test-schemas/userpool-static-dynamic-fields.ts

```diff
@@ -0,0 +1,86 @@
+import { ImportedRDSType } from '@aws-amplify/graphql-transformer-core';
+import { generateDDL } from '../../rds-v2-test-utils';
+
+export const schema = `
+  # Owners may update their owned records.
+  # Admins may create Employee records.
+  # Any authenticated user may view Employee ids, bios & emails.
+  # Owners and members of "Admin" group may see employee salaries.
+  # Owners of "Admin" group may create and update employee salaries.
+  type Employee
+    @model
+    @auth(
+      rules: [
+        { allow: groups, groups: ["Admin"] }
+        { allow: private, operations: [read] }
+        { allow: owner, ownerField: "email", operations: [read, update] }
+      ]
+    ) {
+    id: ID! @primaryKey
```

Comment:
Check if this compiles since I am seeing an error `🛑 When using field-level authorization rules you need to add rules to all of the model's required fields with at least 
read permissions. Found model "Post" with required fields ["id"] missing field-level authorization rules. ` in your other newly added tests `rds-mysql-field-auth-userpool-iam` and `rds-pg-field-auth-userpool-iam`.